modules/Emails/include/ListView/ListViewDataEmails.php

```diff
@@ -116,849 +132,590 @@ public function __construct()
-                    }
+            $folder = $inboundEmail->mailbox;
 
-                    if (!is_array($params)) {
-                        $params = array();
-                    }
-                    if (!isset($params['custom_select'])) {
-                        $params['custom_select'] = '';
-                    }
-                    if (!isset($params['custom_from'])) {
-                        $params['custom_from'] = '';
-                    }
-                    if (!isset($params['custom_where'])) {
-                        $params['custom_where'] = '';
-                    }
-                    if (!isset($params['custom_order_by'])) {
-                        $params['custom_order_by'] = '';
-                    }
+            $filter = $this->getFilter($filter_fields, $where, $request);
 
-                    $this->setVariableName($seed->object_name, $crmWhere, $this->listviewName, $id);
 
-                    $this->seed->id = '[SELECT_ID_LIST]';
+            // carry out the filter type
+            switch ($this->searchType) {
+                case 'crm':
 
-                    $nextOffset = -1;
-                    $prevOffset = -1;
-                    $endOffset = 0;
 
+                    $limit = $sugar_config['list_max_entries_per_page'];
 
-                    if ($total > $limitPerPage) {
-                        $nextOffset = $offset + $limitPerPage;
-                    }
+                    $search = new ListViewDataEmailsSearchOnCrm($this);
+                    $ret = $search->search($filter_fields, $request, $where, $inboundEmail, $params, $seed, $singleSelect, $id, $limit, $current_user, $id_field, $offset);
 
-                    if ($nextOffset >= $total) {
-                        $nextOffset = $total;
-                    }
 
-                    if ($page > 0) {
-                        $prevOffset = $offset - $limitPerPage;
-                        if ($prevOffset < 0) {
-                            $prevOffset = 0;
-                        }
-                    }
 
-                    if ($total < $limitPerPage) {
-                        $prevOffset = -1;
-                        $nextOffset = -1;
-                    }
+                    break;
+                case 'imap':
 
-                    if ($total > 0) {
-                        $endOffset = $total / $limitPerPage;
-                    }
 
-                    $pageData['offsets']['current'] = $offset;
-                    $pageData['offsets']['total'] = $total;
-                    $pageData['offsets']['next'] = $nextOffset;
-                    $pageData['offsets']['prev'] = $prevOffset;
-                    $pageData['offsets']['end'] = ceil($endOffset);
+                    $limitPerPage = isset($sugar_config['list_max_entries_per_page']) && (int)$sugar_config['list_max_entries_per_page'] ? $sugar_config['list_max_entries_per_page'] : 10;
 
```

Comment:
Lots of white space. Perhaps it would be easier to read with out the extra double lines.